File: returned/python/review/longest_consecutive_sequence.py
# https://leetcode.com/problems/longest-consecutive-sequence/
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        longest = 0
        for n in nums_set:
            if n-1 not in nums_set:
                length = 1
                while n+length in nums_set:
                    length += 1
                longest = max(longest, length)
        return longest

File: typescript/today2/longest-consecutive-sequence.ts
/**
 * https://leetcode.com/problems/longest-consecutive-sequence/description/
 */
export { longestConsecutive };

function longestConsecutive(nums: number[]): number {
  const set = new Set(nums);
  let longest = 0;

  for (let n of set) {
    if (!set.has(n - 1)) {
      let j = n + 1;
      while (set.has(j)) {
        j++;
      }
      longest = Math.max(longest, j - n);
    }
  }

  return longest;
}
